lib/engine/part/path.rb

```diff
@@ -98,21 +98,55 @@ def path?
       end
 
       def node?
-        @_node ||= @nodes.any?
+        return @_node if defined?(@_node)
+
+        @_node = @nodes.any?
       end
 
       def terminal?
-        @_terminal ||= !!@terminal
+        return @_terminal if defined?(@_terminal)
```

Comment:
there's no reason to cache this, it's not expensive at all just do

def terminal?
!!@terminal

Reply:
OK.